### PR TITLE
[#3327]: Introduce wallet onboarding screens

### LINKED ISSUE #3327: Introduce wallet onboarding screens
Image

zpl://screen?sid=5a8711ba690d3bba6c2a35b4&pid=5a55d587ba304eed4d2c942

### PR DESCRIPTION
**Pull request by alwx**
Fixes #3327 
Introduces wallet onboarding screens

Status: ready

### CODE REVIEW

File: src/status_im/data_store/realm/schemas/base/v8/account.cljs [hunk 1]
```diff
@@ -15,6 +15,9 @@
                           :photo-path          :string
                           :signing-phrase      {:type :string}
                           :mnemonic            {:type :string}
+                          ;; TODO(alwx): move it to v9.account
```
Comment (flexsurfer) [inline src/status_im/data_store/realm/schemas/base/v8/account.cljs]: @alwx could you elaborate?

Comment (jeluard) [inline src/status_im/data_store/realm/schemas/base/v8/account.cljs]: Do we want to address that now?

File: src/status_im/translations/en.cljs [hunk 2]
```diff
@@ -153,6 +153,14 @@
    :word-n-description                   "In order to check if have backed up your seed phrase correctly, enter the word #{{number}} above"
    :word-n                               "Word #{{number}}"
 
+   :wallet-onboarding-title              "Simple and secure cryptocurrency wallet"
+   :wallet-onboarding-description        "Lorem ipsum"
```
Comment (jeluard) [inline src/status_im/translations/en.cljs]: Lorem ipsum?

File: src/status_im/translations/en.cljs [hunk 3]
```diff
@@ -153,6 +153,14 @@
...
+   :wallet-onboarding-set-up             "Set up my Wallet"
+   :wallet-set-up-title                  "Set up your wallet"
+   :wallet-set-up-signing-phrase         "This is your personal signing phrase. You'll see it everytime you make a transaction. Only confirm a transaction when you see your signing phrase."
+   :wallet-set-up-confirm-title          "Are you sure?"
```
Comment (jeluard) [inline src/status_im/translations/en.cljs]: We already have `:are-you-sure?`

File: src/status_im/ui/components/common/styles.cljs [hunk 4]
```diff
@@ -138,10 +138,10 @@
...
     {:padding-vertical   12
      :padding-horizontal 42
-     :border-radius      8}
-    style
-    (when background?
-      {:background-color (colors/alpha colors/blue 0.1)})))
+     :border-radius      8
+     :background-color   (when background?
+                           (colors/alpha colors/blue 0.1))}
+    style))
```
Comment (flexsurfer) [inline src/status_im/ui/components/common/styles.cljs:144]: what the difference?

File: src/status_im/ui/screens/wallet/onboarding/setup/views.cljs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            [status-im.ui.screens.wallet.components :as comp]
+            [status-im.ui.screens.wallet.onboarding.setup.styles :as styles]
+            [status-im.ui.components.bottom-buttons.view :as bottom-buttons]
+            [status-im.ui.components.button.view :as button]
+            [status-im.utils.utils :as utils]))
+
+(defn signing-word [word]
+  [react/view styles/signing-word
+   [react/text {:style           styles/signing-word-text
+                :font            :roboto-mono
```
Comment (jeluard) [inline src/status_im/ui/screens/wallet/onboarding/setup/views.cljs:18]: Maybe we should consider iOS?

File: src/status_im/ui/screens/views.cljs [hunk 6]
```diff
@@ -126,6 +127,7 @@
                         :browser browser
                         :open-dapp open-dapp
                         :dapp-description dapp-description
+                        :wallet-setup wallet.onboarding.setup/screen
```
Comment (flexsurfer) [inline src/status_im/ui/screens/views.cljs]: confrontation? :)

**Review by jeluard — CHANGES_REQUESTED** [review]

File: src/status_im/ui/screens/wallet/events.cljs [hunk 7]
```diff
@@ -84,6 +85,11 @@
   (fn [{:keys [web3 success-event edit?]}]
     (ethereum/gas-price web3 #(re-frame/dispatch [success-event %2 edit?]))))
 
+(re-frame/reg-fx
+  ::save-account
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/events.cljs]: that's weird we haven't had this fx, are you sure?

Comment (flexsurfer) [reply src/status_im/ui/screens/wallet/events.cljs]: we have this fx in `status-im.ui.screens.accounts.events`

File: src/status_im/ui/screens/wallet/events.cljs [hunk 8]
```diff
@@ -224,3 +230,12 @@
   :wallet/show-error
   (fn []
     {:show-error (i18n/label :t/wallet-error)}))
+
+(handlers/register-handler-fx
+  :wallet/set-up-passed
+  (fn [{{:accounts/keys [accounts current-account-id] :as db} :db}]
+    (let [current-account (get accounts current-account-id)
+          new-account (assoc current-account :wallet-set-up-passed? true)]
+      {:db            (assoc-in db [:accounts/accounts current-account-id] new-account)
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/events.cljs]: you can use `status-im.ui.screens.accounts.events/account-update` it prepares fx

File: src/status_im/ui/screens/wallet/onboarding/setup/views.cljs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        [react/view {:style styles/signing-phrase}
+         (for [word signing-words]
+           ^{:key (str "signing-word-" word)}
+           [signing-word word])]
+        [react/text {:style styles/description}
+         (i18n/label :t/wallet-set-up-signing-phrase)]
+        [bottom-buttons/bottom-buttons styles/bottom-buttons
+         nil
+         [button/button {:on-press            display-confirmation
+                         :text-style          {:padding-horizontal 0}
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/onboarding/setup/views.cljs]: move to styles?

File: src/status_im/ui/screens/wallet/onboarding/views.cljs [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  [react/view styles/root
+   [react/view {:style styles/onboarding-image-container}
+    [react/image {:source (:wallet-welcome resources/ui)
+                  :style  styles/onboarding-image}]]
+   [react/text {:style styles/onboarding-title}
+    (i18n/label :t/wallet-onboarding-title)]
+   [react/text {:style styles/onboarding-text}
+    (i18n/label :t/wallet-onboarding-description)]
+   [components.common/button
+    {:button-style {:flex-direction   :row
```
Comment (flexsurfer) [inline src/status_im/ui/screens/wallet/onboarding/views.cljs]: move to styles?

**Review by flexsurfer — CHANGES_REQUESTED** [review]

File: src/status_im/ui/components/common/styles.cljs (hunk 4 above)
Comment (alwx) [reply src/status_im/ui/components/common/styles.cljs:144]: the biggest difference is that now it's also possible to provide any other `background-color`

File: src/status_im/ui/screens/wallet/onboarding/setup/views.cljs (hunk 5 above)
Comment (alwx) [reply src/status_im/ui/screens/wallet/onboarding/setup/views.cljs:18]: `RobotoMono-Regular` should also used on iOS (according to the existing designs)

**Comment by alwx** [comment]
Was updated. Please, check it @flexsurfer @jeluard

**Review by flexsurfer — APPROVED** [review]

**Review by jeluard — APPROVED** [review]

**Comment by yevh-berdnyk** [comment]
Android: https://i.diawi.com/fuBZsy
iOS: https://i.diawi.com/hPVuzt

**Bugs found**

When recovering an account a user is forced to set up wallet again and new transaction phrase is generated however it's not updated on previously signed in device. Having different transaction phrases on different devices for one account will confuse a user. It would be better to set up wallet only once when creating an account and just to remind the transaction phrase when recovering the account

**Steps to reproduce:**

On device 1:
- Create new account
- Click Wallet button
- Set up your wallet (remember transaction phrase)

On device 2:
- Recover that account
- Click Wallet button
- Set up your wallet again (remember new transaction phrase)

On device 1:
- Relaunch the app and relogin
- Proceed to sign in a transaction and check the transaction phrase




<blockquote><div><strong><a href="https://i.diawi.com/fuBZsy">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/hPVuzt">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by alwx** [comment]
@yevh-berdnyk please, check the new version

**Comment by yevh-berdnyk** [comment]
Android: https://i.diawi.com/xnau8V
iOS: https://i.diawi.com/Yei8JB

Issues found:

1. **Crash when setting wallet for newly created account**
Reproduction:
  - Create new account
  - Open wallet
  - Click LET’S GET SET UP button
  - Click GOT IT button
  - Click YES button 

After that you need to relaunch the app and wallet is still offering to set up transaction phrase
aa
  

2. **When recovering account of user which has not already set up his wallet a transaction phrase is already changed**

  Reproduction:
On device 1:
  - Create new account
  - Navigate to Wallet and remember a transaction phrase (but don’t set up wallet)
 
On device 2:
  - Recover created account
  - Click Wallet
  - Click Send Transaction
  - Select a recipient and input some amount
  - Click Sign transaction
  - Transaction phrase is changed


<blockquote><div><strong><a href="https://i.diawi.com/xnau8V">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://i.diawi.com/Yei8JB">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Pull request closed without merging**